src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java

```diff
@@ -0,0 +1,99 @@
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.engine.command.Command;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult;
+
+import static edu.harvard.iq.dataverse.engine.command.CommandHelper.CH;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+@RequiredPermissions(Permission.PublishDataset)
+public abstract class AbstractSubmitToArchiveCommand implements Command<DatasetVersion> {
+
+    private final DatasetVersion version;
+    private final DataverseRequest request;
+    private final Map<String, DvObject> affectedDvObjects;
+    private final Map<String, String> requestedSettings = new HashMap<String, String>();
+    private static final Logger logger = Logger.getLogger(AbstractSubmitToArchiveCommand.class.getName());
+    
+    public AbstractSubmitToArchiveCommand(DataverseRequest aRequest, DatasetVersion version) {
+        this.request = aRequest;
+        this.version = version;
+        affectedDvObjects = new HashMap<>();
+        affectedDvObjects.put("", version.getDataset());
+    }
+
+    @Override
+    public DatasetVersion execute(CommandContext ctxt) throws CommandException {
+
+        String settings = ctxt.settings().getValueForKey(SettingsServiceBean.Key.ArchiverSettings);
+        String[] settingsArray = settings.split(",");
+        for (String setting : settingsArray) {
+            setting = setting.trim();
+            if (!setting.startsWith(":")) {
+                logger.warning("Invalid Archiver Setting: " + setting);
+            } else {
+                requestedSettings.put(setting, ctxt.settings().get(setting));
+            }
+        }
+        
+        AuthenticatedUser user = request.getAuthenticatedUser();
+        ApiToken token = ctxt.authentication().findApiTokenByUser(user);
+        if ((token == null) || (token.getExpireTime().before(new Date()))) {
+            token = ctxt.authentication().generateApiTokenForUser(user);
+        }
+        performArchiveSubmission(version, token, requestedSettings);
+        return ctxt.em().merge(version);
+    }
+
+    @Override
+    public Map<String, DvObject> getAffectedDvObjects() {
+
+        return affectedDvObjects;
+    }
+
+    @Override
+    public DataverseRequest getRequest() {
+        return request;
+    }
+
+    @Override
+    public Map<String, Set<Permission>> getRequiredPermissions() {
+        return CH.permissionsRequired(getClass());
+    }
+
+    @Override
+    public String describe() {
```

Comment:
@qqmyers this is used by the ActionLogRecord; there is a generic implementation in AbstractCommand.

Reply:
OK - how's my update to more closely mirror the format for DvObjects in AbstractCommand (latest commit)?

Reply:
It's ok to make this similar to the abstractCommand (though I'm still hoping you won't need to) or to modify it to put info that is important to record in the ActionLogRecord, but not handled by the generic describe. So, I'm ok with this.